week3/homework/src/index.js

```diff
@@ -1,3 +1,137 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+{
+  const express = require('express');
+  const uuid = require('uuid');
+  const fs = require('fs');
+
+  const app = express();
+
+  app.use(express.json());
+
+  function readTodosFromFile() {
+    return new Promise((resolve, reject) => {
+      fs.readFile('todos.json', 'utf8', (err, data) => {
+        if (err) reject(err);
+        else resolve(JSON.parse(data));
+      });
+    });
+  }
+
+  function saveTodos(todos) {
+    return new Promise((resolve, reject) => {
+      fs.writeFile('todos.json', JSON.stringify(todos), 'utf8', err => {
+        if (err) reject(err);
+        else resolve();
+      });
+    });
+  }
+
+  function parseAndValidate(request) {
+    const { todo } = request.body;
+    if (todo == null) throw Error;
+    if (todo.description != null) todo.description = todo.description.trim();
+    if (todo.description == null || todo.description.length < 3) throw Error;
+    return todo;
+  }
+
+  async function readTodos(req, res) {
+    const todos = await readTodosFromFile();
+    res.send(todos);
+  }
+
+  // Add a todo and todo.description in req.body to create todos
+  async function createTodo(req, res) {
+    const todo = parseAndValidate(req);
+    const todos = await readTodosFromFile();
+    todo.id = uuid.v4();
+    todo.isDone = false;
+    todos.push(todo);
+    await saveTodos(todos);
+    markAsNotDone(req, res);
```

Comment:
Can you explain why are you calling markAsNotDone here?

Reply:
Oops I forgot to delete that thanks for telling me:) I was trying to make todos 'not done' when they were created. At first I tried to use this function but then I said 'What am I doing? There is a much simpler way.'  And so I forgot to delete that line.